### PR TITLE
Bug fix: for compare known, on windows, use "\" instead of "/"

### PR DESCRIPTION
**Pull request by c0ffeec0der**
Bug fix: for compare known, on windows, use "\\" instead of "/"

to solve issue https://github.com/coreos/kube-prometheus/issues/557

### CODE REVIEW

**Comment by c0ffeec0der** [comment]
Debug result for the function "known"

Before

p .tmp, k github.com/kubernetes-monitoring/kubernetes-mixin\lib\promgrafonnet
p .tmp, k github.com/metalmatze/slo-libsonnet\slo-libsonnet
p .tmp, k github.com/coreos/kube-prometheus/jsonnet/kube-prometheus
p .tmp, k github.com/kubernetes-monitoring/kubernetes-mixin
p .tmp, k github.com/coreos/prometheus-operator\jsonnet\prometheus-operator
p .tmp, k github.com/grafana/grafonnet-lib\grafonnet
p .tmp, k github.com/prometheus/prometheus\documentation\prometheus-mixin
p .tmp, k github.com/ksonnet/ksonnet-lib
p .tmp, k github.com/coreos/etcd\Documentation\etcd-mixin
p .tmp, k github.com/prometheus/node_exporter\docs\node-mixin
p .tmp, k github.com/grafana/jsonnet-libs\grafana-builder
p .tmp, k github.com/brancz/kubernetes-grafana\grafana
p github.com, k github.com/kubernetes-monitoring/kubernetes-mixin
p github.com/brancz, k github.com/coreos/etcd\Documentation\etcd-mixin
p github.com/brancz, k github.com/prometheus/node_exporter\docs\node-mixin
p github.com/brancz, k github.com/grafana/jsonnet-libs\grafana-builder
p github.com/brancz, k github.com/brancz/kubernetes-grafana\grafana
p github.com/brancz/kubernetes-grafana, k github.com/kubernetes-monitoring/kubernetes-mixin\lib\promgrafonnet
p github.com/brancz/kubernetes-grafana, k github.com/metalmatze/slo-libsonnet\slo-libsonnet
p github.com/brancz/kubernetes-grafana, k github.com/coreos/kube-prometheus/jsonnet/kube-prometheus
p github.com/brancz/kubernetes-grafana, k github.com/kubernetes-monitoring/kubernetes-mixin
p github.com/brancz/kubernetes-grafana, k github.com/coreos/prometheus-operator\jsonnet\prometheus-operator
p github.com/brancz/kubernetes-grafana, k github.com/grafana/grafonnet-lib\grafonnet
p github.com/brancz/kubernetes-grafana, k github.com/prometheus/prometheus\documentation\prometheus-mixin
p github.com/brancz/kubernetes-grafana, k github.com/ksonnet/ksonnet-lib
p github.com/brancz/kubernetes-grafana, k github.com/coreos/etcd\Documentation\etcd-mixin
p github.com/brancz/kubernetes-grafana, k github.com/prometheus/node_exporter\docs\node-mixin
p github.com/brancz/kubernetes-grafana, k github.com/grafana/jsonnet-libs\grafana-builder
p github.com/brancz/kubernetes-grafana, k github.com/brancz/kubernetes-grafana\grafana
p github.com/brancz/kubernetes-grafana/grafana, k github.com/grafana/jsonnet-libs\grafana-builder
p github.com/brancz/kubernetes-grafana/grafana, k github.com/brancz/kubernetes-grafana\grafana
p github.com/brancz/kubernetes-grafana/grafana, k github.com/prometheus/prometheus\documentation\prometheus-mixin
p github.com/brancz/kubernetes-grafana/grafana, k github.com/ksonnet/ksonnet-lib
p github.com/brancz/kubernetes-grafana/grafana, k github.com/coreos/etcd\Documentation\etcd-mixin
p github.com/brancz/kubernetes-grafana/grafana, k github.com/prometheus/node_exporter\docs\node-mixin
p github.com/brancz/kubernetes-grafana/grafana, k github.com/coreos/prometheus-operator\jsonnet\prometheus-operator
p github.com/brancz/kubernetes-grafana/grafana, k github.com/grafana/grafonnet-lib\grafonnet
p github.com/brancz/kubernetes-grafana/grafana, k github.com/kubernetes-monitoring/kubernetes-mixin\lib\promgrafonnet
p github.com/brancz/kubernetes-grafana/grafana, k github.com/metalmatze/slo-libsonnet\slo-libsonnet
p github.com/brancz/kubernetes-grafana/grafana, k github.com/coreos/kube-prometheus/jsonnet/kube-prometheus
p github.com/brancz/kubernetes-grafana/grafana, k github.com/kubernetes-monitoring/kubernetes-mixin

After

p .tmp, k github.com\coreos\kube-prometheus\jsonnet\kube-prometheus
p .tmp, k github.com\prometheus\prometheus\documentation\prometheus-mixin
p .tmp, k github.com\metalmatze\slo-libsonnet\slo-libsonnet
p .tmp, k github.com\coreos\prometheus-operator\jsonnet\prometheus-operator
p .tmp, k github.com\grafana\jsonnet-libs\grafana-builder
p .tmp, k github.com\grafana\grafonnet-lib\grafonnet
p .tmp, k github.com\prometheus\node_exporter\docs\node-mixin
p .tmp, k github.com\kubernetes-monitoring\kubernetes-mixin\lib\promgrafonnet
p .tmp, k github.com\coreos\etcd\Documentation\etcd-mixin
p .tmp, k github.com\ksonnet\ksonnet-lib
p .tmp, k github.com\kubernetes-monitoring\kubernetes-mixin
p .tmp, k github.com\brancz\kubernetes-grafana\grafana
p github.com, k github.com\coreos\kube-prometheus\jsonnet\kube-prometheus
p github.com\brancz, k github.com\metalmatze\slo-libsonnet\slo-libsonnet
p github.com\brancz, k github.com\coreos\kube-prometheus\jsonnet\kube-prometheus
p github.com\brancz, k github.com\prometheus\prometheus\documentation\prometheus-mixin
p github.com\brancz, k github.com\ksonnet\ksonnet-lib
p github.com\brancz, k github.com\kubernetes-monitoring\kubernetes-mixin
p github.com\brancz, k github.com\coreos\prometheus-operator\jsonnet\prometheus-operator
p github.com\brancz, k github.com\grafana\jsonnet-libs\grafana-builder
p github.com\brancz, k github.com\grafana\grafonnet-lib\grafonnet
p github.com\brancz, k github.com\prometheus\node_exporter\docs\node-mixin
p github.com\brancz, k github.com\kubernetes-monitoring\kubernetes-mixin\lib\promgrafonnet
p github.com\brancz, k github.com\coreos\etcd\Documentation\etcd-mixin
p github.com\brancz, k github.com\brancz\kubernetes-grafana\grafana
p github.com\brancz\kubernetes-grafana, k github.com\coreos\kube-prometheus\jsonnet\kube-prometheus
p github.com\brancz\kubernetes-grafana, k github.com\prometheus\prometheus\documentation\prometheus-mixin
p github.com\brancz\kubernetes-grafana, k github.com\metalmatze\slo-libsonnet\slo-libsonnet
p github.com\brancz\kubernetes-grafana, k github.com\grafana\jsonnet-libs\grafana-builder
p github.com\brancz\kubernetes-grafana, k github.com\grafana\grafonnet-lib\grafonnet
p github.com\brancz\kubernetes-grafana, k github.com\prometheus\node_exporter\docs\node-mixin
p github.com\brancz\kubernetes-grafana, k github.com\kubernetes-monitoring\kubernetes-mixin\lib\promgrafonnet
p github.com\brancz\kubernetes-grafana, k github.com\coreos\etcd\Documentation\etcd-mixin
p github.com\brancz\kubernetes-grafana, k github.com\ksonnet\ksonnet-lib
p github.com\brancz\kubernetes-grafana, k github.com\kubernetes-monitoring\kubernetes-mixin

**Comment by brancz** [comment]
@theSuess you recently worked on windows support. Could you double check this one?

File: pkg/packages.go [hunk 1]
```diff
@@ -200,16 +204,26 @@ func checkLegacyNameTaken(legacyName string, pkgName string) (bool, error) {
 }
 
 func known(deps map[string]deps.Dependency, p string) bool {
-	p = filepath.ToSlash(p)
+	p = Subdir(filepath.ToSlash(p))
 	for _, d := range deps {
-		k := d.Name()
+		k := Subdir(d.Name())
```
Comment (theSuess) [inline pkg/packages.go:209]:
```suggestion
		k := filepath.ToSlash(d.Name())
```
This change alone would suffice to fix the issue.

In your change, every `/` in `p` is replaced by `\` and the same for `d.Name()`. Using `filepath.ToSlash` we can eliminate the need for the `Subdir` function

Comment (c0ffeec0der) [reply pkg/packages.go]: Ah okay, I have just pushed the updated code. Let me know if there is anything i miss

**Comment by c0ffeec0der** [comment]
Request for re-review

**Comment by theSuess** [comment]
LGTM

**Comment by davidovich** [comment]
LGTM

**Pull request merged**